.github/workflows/pytest.yml

```
@@ -0,0 +1,25 @@
+name: Pytest PR
+on:
+  pull-request:
+    branches:
+      - main
+jobs:
+  run-test-code:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

`requirements.txt` is required for this pipeline. So this condition is not necessary here.

```suggestion
         pip install -r requirements.txt
```